distributed/scheduler.py

```diff
@@ -3020,14 +2984,15 @@ def transition_released_forgotten(self, key, stimulus_id):
         ("waiting", "no-worker"): transition_waiting_no_worker,
         ("waiting", "queued"): transition_waiting_queued,
         ("waiting", "memory"): transition_waiting_memory,
+        ("queued", "memory"): partial(impossible_transition, "memory"),
         ("queued", "released"): transition_queued_released,
         ("queued", "processing"): transition_queued_processing,
         ("processing", "released"): transition_processing_released,
         ("processing", "memory"): transition_processing_memory,
         ("processing", "erred"): transition_processing_erred,
         ("no-worker", "released"): transition_no_worker_released,
         ("no-worker", "processing"): transition_no_worker_processing,
-        ("no-worker", "memory"): transition_no_worker_memory,
+        ("no-worker", "memory"): partial(impossible_transition, "memory"),
```

Comment:
What's the value of having a transition function for an impossible transition, versus updating
https://github.com/dask/distributed/blob/5a14053b152bf73960afc5d0f3768bc70bd5124c/distributed/scheduler.py#L1853
to raise a more detailed error like you have here (including the story)?

Reply:
The idea was to avoid the error from being silenced in the future if anybody adds a (released, memory) transition. But in hindsight I'll just overhaul the above.